x-pack/plugins/apm/server/routes/metrics/get_metrics_chart_data_by_agent.ts

```diff
@@ -29,25 +31,26 @@ export async function getMetricsChartDataByAgent({
   agentName: string;
   start: number;
   end: number;
+  serviceRuntimeName?: string;
 }): Promise<GenericMetricsChart[]> {
-  if (isJavaAgentName(agentName)) {
-    return getJavaMetricsCharts({
-      environment,
-      kuery,
-      setup,
-      serviceName,
-      serviceNodeName,
-      start,
-      end,
-    });
-  }
-
-  return getDefaultMetricsCharts({
+  const options = {
     environment,
     kuery,
     setup,
     serviceName,
     start,
     end,
-  });
+  };
+  if (isJavaAgentName(agentName)) {
+    return getJavaMetricsCharts({
+      ...options,
+      serviceNodeName,
+    });
+  }
+
+  if (isServerlessAgent(serviceRuntimeName)) {
```

Comment:
Actually, Java metrics and serverless metrics are not exclusive of each other (i.e. means we could have both). Since, for now we disabled agent metrics anyways it's fine for now. But later we'll need to be able to enable both (e.g. by having them as different tabs in the infrastructure / runtime metrics tab view).

Reply:
> I guess this will be resolved with @boriskirov's redesign of the metrics tabs.

Yes, that's why this PR only contains API changes, not UI.